File: backend/src/main/java/com/example/backend/model/Wishlist.java
package com.example.backend.model;

import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@With
@Document("wishlists")
public record Wishlist(
        @Id
        String wishlistId,
        String name,
        List<Wish>wishes
        ) {
}

File: frontend/src/model/Wishlist.ts
/*interface NameIdInterface {
    id: string,
    name: string,
}

export interface Wishlist extends NameIdInterface {
    wishes: Array<Wish>
}
export interface Wish extends NameIdInterface {
    status: Status
}*/


import {Wish} from "./Wish";

export type Wishlist = {
    wishlistId?: string,
    name: string,
    wishes: Wish[]
}